src/components/Resource/ResourceCreate.tsx

```diff
@@ -32,48 +32,48 @@ import request from "@/Utils/request/request";
 import useQuery from "@/Utils/request/useQuery";
 import { parsePhoneNumber } from "@/Utils/utils";
 
+import CircularProgress from "../Common/CircularProgress";
+import UserAutocomplete from "../Common/UserAutocompleteFormField";
+import { FacilityModel } from "../Facility/models";
+import Form from "../Form/Form";
+import { UserBareMinimum } from "../Users/models";
+
 interface resourceProps {
   facilityId: number;
+  resourceId?: string;
+}
+
+interface ResourceData {
```

Comment:
See if you can use ResourceModel instead of defining another data structure here.

Probably will have to use Partial<ResourceModel>